docs/user-guide/app_dependencies.md

```diff
@@ -0,0 +1,133 @@
+* Syncing an application that is dependent on other apps will not trigger a sync of the dependencies. However, manual sync of a dependent application whose dependencies are not satisfied will block unless force sync is used.
+* Dependencies currently can only be configured in a declarative manner; the UI and CLI do not yet have first class support for configuring dependencies.
+
+It should be noted that at this time, dependencies will have no effect on Application deletion. That is, Argo CD will allow you to delete an Application which other Applications still depend on. This is likely to change in a future iteration of this feature.
+
+!!!warning
+    Using force sync to override application dependencies should only be used as a last resort, as it may break your applications in multiple ways.
+
+## Configuring Application dependencies
+
+The dependencies of an Application are configured using the field `dependsOn` in the Application's spec. The field `dependsOn` is a complex data type, which lets you define one or more `selectors` to match a set of other Applications that this Application will be dependent upon.
+
+Each selector can match Applications according to one or more properties. If you specify more than one matcher in a selector, each matcher has to match in order for an Application to be considered a dependency. For example, if you specify a `labelSelector` and a `namePattern`, both will have to match in order for an Application to be selected.
+
+Additionally, the behaviour of dependency resolution can be configured on a per Application basis.
+
+## Dependency selectors
+
+Currently, the available properties to match against are `labels` and the `name` of the Application. Each selector can yield zero or more applications.
+
+### Matching on labels
+
+It is possible and recommended to use a standard label selector to select the dependencies for your Application:
+
+```yaml
+spec:
+  dependsOn:
+    selectors:
+    - labelSelector:
+        matchLabels:
+            name: some-app
+```
+
+The above YAML will have the Application depend on any other Application that matches the given label selector. In this case, any application carrying a label `name` with the value of `some-app`.
+
+### Matching on name patterns
+
+Dependencies can be matched on the name of Applications by specifying the `namePattern` selector. The `namePattern` selector takes a list of Application name patterns to match against the Applications' `.metadata.name` property. Specifying multiple patterns 
+
+```yaml
+spec:
+  dependsOn:
+    selectors:
+    - namePattern:
+      - some-app
+      - other-app
+```
+
+The above YAML will have the Application depend on the Applications `some-app` and `other-app`. 
+
+The `namePattern` matcher supports shell-style glob matching, so the following example would match all Applications whose name is starting with `stage1-`:
```

Comment:
Can we specify the glob engine used to evaluate these and link to its docs? We use multiple glob engines for different purposes, so being specific should be quite helpful to the user.